pkg/restapi/operation/didexchange/didexchange.go

```diff
@@ -217,10 +210,26 @@ func (c *Operation) QueryConnections(rw http.ResponseWriter, req *http.Request)
 		},
 	}
 
-	err = json.NewEncoder(rw).Encode(response)
+	c.writeResponse(rw, response)
+}
+
+// getQueryParams unmarshals query strings values to interface value
+func (c *Operation) getQueryParams(v interface{}, vals url.Values) error {
+
+	// normalize all query string key/values
+	args := make(map[string]string)
+	for k, v := range vals {
+		if len(v) > 0 {
+			args[k] = v[0]
+		}
+	}
+
+	bytes, err := json.Marshal(args)
 	if err != nil {
-		logger.Errorf("Unable to write response, %s", err)
+		return err
 	}
+
+	return json.Unmarshal(bytes, v)
```

Comment:
Interesting pattern. I wonder if the Go stdlib has a more efficient method (rather than marshal and unmarshal json).

Reply:
Using Go standard lib supports, 
can convert 'map[string]string' to struct with string properties.
can convert 'map[string]string[]' to struct with string[] properties.

In this case, net.http `request.URL.Query()` was returning `map[string]string[]` and we needed struct with string properties out of it.

I didn't want to manually lookup for `key,value` pair inside `request.URL.Query()` to set them one by one to target struct (hard to maintain whenever queryString/model changes and developers may forgot update this logic whenever they change models).

So went for dynamic solution.

Reply:
Right - I guess I was surprised that the only way to accomplish the above (dynamically) is via JSON marshal/unmarshal.